hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotManager.java

```diff
@@ -670,11 +670,13 @@ public void takeSnapshot(SnapshotDescription snapshot) throws IOException {
     }
   }
 
-  public synchronized long takeSnapshot(SnapshotDescription snapshot, long nonceGroup, long nonce)
+  public long takeSnapshot(SnapshotDescription snapshot, long nonceGroup, long nonce)
     throws IOException {
     this.takingSnapshotLock.readLock().lock();
     try {
-      return submitSnapshotProcedure(snapshot, nonceGroup, nonce);
+      synchronized(this){
```

Comment:
I think we can just add a synchronized modifier on the submitSnapshotProcedure method? To align with other methods in the class?

Reply:
Thank you for your review and valuable suggestions. I have made the necessary changes based on the suggestions provided.